packages/app-mobile/components/NoteList.tsx

```diff
@@ -0,0 +1,263 @@
+
+				if (doIt) {
+					Setting.setValue('notes.sortOrder.field', 'order');
+				}
+			};
+		}
+
+		const noteContent = (
+			<View style={this.styles().noteContainer}>
+				<NoteItem
+					note={note}
+					onLongPress={drag}
+					disabled={additionalProps && additionalProps.isActive}
+				/>
+			</View>
+		);
+
+		if (this.props.noteSelectionEnabled) {
+			return noteContent;
+		}
+
+		return (
+			<Swipeable renderLeftActions={renderLeftActions} onSwipeableOpen={onOpen} ref={(swipeable) => currentSwipeable = swipeable}>
+				{noteContent}
+			</Swipeable>
+		);
+	}
+
+	public renderMainContent() {
+		// `enableEmptySections` is to fix this warning: https://github.com/FaridSafi/react-native-gifted-listview/issues/39
+
+		if (this.props.items.length) {
+			return (
+				<View style={{ flex: 1 }}>
+					<DraggableFlatList
+						ref={(ref: any) => (this.rootRef_ = ref)}
+						data={this.state.items}
+						renderItem={({ item, drag, isActive }) => (<ScaleDecorator>
+							{this.renderNoteItem(item, { drag, isActive })}
+						</ScaleDecorator>)}
+						keyExtractor={item => item.id}
+						onDragEnd={async ({ data, to }) => {
+							if (this.props.selectedFolderId) {
+								this.setState({ items: data });
+								await Note.insertNotesAt(
+									this.props.selectedFolderId,
+									[data[to].id],
+									to,
+									Setting.value('uncompletedTodosOnTop'),
+									Setting.value('showCompletedTodos')
+								);
+								this.props.dispatch({
+									type: 'NOTE_UPDATE_ALL',
+									notes: data,
+									notesSource: this.props.notesSource,
+								});
+							}
+						}}
+					/>
+				</View>
+			);
+		} else {
+			if (!this.props.folders.length) {
+				const noItemMessage = _('You currently have no notebooks.');
+				return (
+					<View style={this.styles().noNotebookView}>
+						<Text style={this.styles().noItemMessage}>{noItemMessage}</Text>
+						<Button title={_('Create a notebook')} onPress={this.createNotebookButton_click} />
+					</View>
+				);
+			} else {
+				const noItemMessage = _('There are currently no notes. Create one by clicking on the (+) button.');
+				return <Text style={this.styles().noItemMessage}>{noItemMessage}</Text>;
+			}
+		}
+	}
+	public render() {
+		return (
+			<>
+				{this.renderMainContent()}
+				<DialogBox
+					ref={(dialogbox: any) => {
+						this.dialogbox = dialogbox;
+					}}
+				/>
+			</>
+		);
+	}
+}
+
+const NoteList = connect((state: AppState) => {
+	return {
+		items: state.notes,
+		folders: state.folders,
+		selectedFolderId: state.selectedFolderId,
+		notesSource: state.notesSource,
+		themeId: state.settings.theme,
+		noteSelectionEnabled: state.noteSelectionEnabled,
+	};
+})(NoteListComponent);
+
+export default NoteList;
```

Comment:
Unfortunately I can't review this, since git shows this as a deleted file and a new file instead of a renamed file. You may have to use the `git mv` command or similar to make it work. If we can't review we can't merge

Reply:
Ok I was hoping `mv` would be part of the heuristic they could use for this, but indeed if it's not recorded anywhere it's annoying. There isn't much that's changed in this file actually, so I'm surprised git cannot detect that it's the same file. Perhaps we should have two pull requests - one for the TS conversion, and one for the actual changes

Reply:
Thanks for the explanation @TaoK, I guess perhaps because it's a different file type it's not detected as a rename? I can separate out the TS change.

Reply:
> I guess perhaps because it's a different file type it's not detected as a rename?

No, the rename detection is based on file change % (by lines). The default change % to trigger a rename detection is 50% - I expect that's what github uses.

> There isn't much that's changed in this file actually

I see most of it changed: I can see it as a rename, locally, by telling git to drop the similarity requirement to 10% rather than the 50% default, eg `git diff ...jcgurango/feature/mobile-sorting --find-renames=10%`, and I then see that almost all the imports are different, the interfaces are new, the declarations are different, much/most of the rendering logic is changed, and the export is different.